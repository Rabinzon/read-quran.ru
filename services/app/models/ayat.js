module.exports = (sequelize, DataTypes) => {
  const Ayat = sequelize.define('Ayat', {
    text: {
      type: DataTypes.STRING(5000),
      allowNull: false,
    },
    order: {
      type: DataTypes.INTEGER,
    },
    surah: {
      type: DataTypes.INTEGER,
    },
    chronology: {
      type: DataTypes.INTEGER,
    },
  }, { timestamps: false });

  Ayat.associate = () => {
  };
  return Ayat;
};
