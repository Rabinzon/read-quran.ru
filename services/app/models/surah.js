module.exports = (sequelize, DataTypes) => {
  const Surah = sequelize.define('Surah', {
    name: {
      type: DataTypes.STRING,
      allowNull: false,
      unique: true,
    },
    chronology: {
      type: DataTypes.INTEGER,
    },
  }, { timestamps: false });

  Surah.associate = (models) => {
    models.Surah.hasMany(models.Ayat, { foreignKey: 'surah' });
  };
  return Surah;
};
