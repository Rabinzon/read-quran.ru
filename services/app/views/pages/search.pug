extends /layouts/application.pug

block content
  h5.mt-4 Количество найденных аятов по запросу&nbsp;
    i: strong!=q + ':&nbsp;'
    !=results.length
  .mb-4: small.text-muted Аяты отсортированы в хронологическом порядке.

  .row
    .col: .pl-0
      each ayat in results || []
        p.my-1
          span.badge.badge-dark=ayat.surah + ':' + ayat.order
          a.p-1.text-muted.font-normal(href=urlFor('surah', ayat.chronology) + '#' + ayat.order)!=ayat.result
