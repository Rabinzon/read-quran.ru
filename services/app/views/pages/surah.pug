extends /layouts/application.pug
include /mixins/pagination.pug

block content
  .row.justify-content-md-center
    .col-md-8.col-sm-12
      .my-3
        h1(style="word-break: break-all;").mb-0: small= 'Сура ' + surah.id + ' «' + surah.name + '»'
        small: i хронолоческий порядок суры &nbsp;
          !=surah.chronology
        - var url = isCanonical ? urlFor('canonical', surah.id) : urlFor('surah', surah.chronology);
        if translator === 'krachkovsky'
          p: a(href= url + '/kuliev') Читать суру в переводе Э. Р. Кулиева
        else
          p: a(href=url) Читать суру в переводе И. Ю. Крачковского
  .row
    .col-md-8.col-sm-12.m-auto
      each ayat in surah.Ayats || []
        .ayat.d-flex
          .d-block: .badge.badge-dark=ayat.surah + ':' + ayat.order
          .p-1(id=ayat.order)!=ayat.text
      +pagination(surah)
