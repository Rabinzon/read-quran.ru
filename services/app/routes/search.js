import models from '../models';

export default (router) => {
  router.get('search', '/search', async (ctx) => {
    const { q } = ctx.query;
    const results = await models.sequelize.query(`
      SELECT surah, "order", "chronology", ts_headline(text, keywords, 'StartSel=<mark>,StopSel=</mark>,HighlightAll=true') as result
      FROM "Ayats", plainto_tsquery('russian', :query) as keywords
      WHERE TRANSLATOR='krachkovsky' AND _search @@ keywords ORDER BY "chronology";
    `, {
      model: models.Ayats,
      replacements: { query: q },
    });
    ctx.render('pages/search', { results: results[0], q, title: `Поиск по запросу: ${q}` });
  });
};
