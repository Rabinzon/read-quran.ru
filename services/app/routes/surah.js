import _ from 'lodash';
import { Surah, Ayat } from '../models';

export default (router) => {
  router
    .get('surah', '/s/:id/:translator?', async (ctx) => {
      const translator = ctx.params.translator || 'krachkovsky';

      const surah = await Surah.find({
        where: {
          chronology: ctx.params.id,
        },
        include: [{
          model: Ayat,
          where: {
            translator,
          },
        }],
        order: [
          [Ayat, 'order', 'ASC'],
        ],
      });

      if (_.isNull(surah)) {
        ctx.throw(404);
        return;
      }

      const surahs = await Surah.findAll({
        order: [
          ['chronology', 'ASC'],
        ],
      });

      const translatorName = translator === 'krachkovsky' ? 'И. Ю. Крачковского' : 'Э. Р. Кулиева';
      const pageTitle = `Сура ${surah.id + 1}. ${surah.name}, перевод ${translatorName}`;

      ctx.render('pages/surah', {
        surah,
        surahs,
        translator,
        isCanonical: false,
        title: pageTitle,
      });
    })
    .get('canonical', '/c/:id/:translator?', async (ctx) => {
      const translator = ctx.params.translator || 'krachkovsky';

      const surah = await Surah.find({
        where: {
          id: ctx.params.id,
        },
        include: [{
          model: Ayat,
          where: {
            translator,
          },
        }],
        order: [
          [Ayat, 'order', 'ASC'],
        ],
      });

      if (_.isNull(surah)) {
        ctx.throw(404);
        return;
      }

      const surahs = await Surah.findAll({
        order: [
          ['id', 'ASC'],
        ],
      });

      const translatorName = translator === 'krachkovsky' ? 'И. Ю. Крачковского' : 'Э. Р. Кулиева';
      const pageTitle = `Сура ${surah.id + 1}. ${surah.name}, перевод ${translatorName}`;

      ctx.render('pages/surah', {
        surah,
        surahs,
        translator,
        isCanonical: true,
        title: pageTitle,
      });
    })
    .get('krack', '/krac/:id', async (ctx) => {
      ctx.redirect(router.url('surah', ctx.params.id));
    });
};
