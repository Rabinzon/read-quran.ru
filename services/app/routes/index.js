import welcome from './welcome';
import surah from './surah';
import search from './search';

const controllers = [
  welcome, surah, search,
];

export default (router, container) => controllers.forEach(f => f(router, container));
