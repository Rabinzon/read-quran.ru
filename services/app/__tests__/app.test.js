import request from 'supertest';
import sequelizeFixtures from 'sequelize-fixtures';
import matchers from 'jest-supertest-matchers';
import models, { sequelize } from '../models';
import ayat from './__fixtures__/ayat.json';
import surah from './__fixtures__/surah.json';
import search from '../migrations/3.search';

import app from '..';

let server;

describe('app', () => {
  beforeAll(async () => {
    jasmine.addMatchers(matchers);

    await sequelize.sync({ force: true });

    await sequelizeFixtures.loadFixtures([
      { model: 'Surah', data: surah },
      { model: 'Ayat', data: ayat },
    ], models);

    await search.up(sequelize.queryInterface);

    server = app().listen(3000);
  });

  describe('requests', () => {
    it('GET index page', async () => {
      const res = await request.agent(server)
        .get('/');

      expect(res).toHaveHTTPStatus(200);
      expect(res.text).toMatchSnapshot();
    }, 10000);

    it('GET surah page', async () => {
      const res = await request.agent(server)
        .get('/s/1');
      expect(res).toHaveHTTPStatus(200);
      expect(res.text).toMatchSnapshot();
    });

    it('GET canonical surah page', async () => {
      const res = await request.agent(server)
        .get('/c/96');
      expect(res).toHaveHTTPStatus(200);
      expect(res.text).toMatchSnapshot();
    });

    it('GET search page', async () => {
      const res = await request.agent(server)
        .get('/search')
        .query({ q: 'читай' });

      expect(res).toHaveHTTPStatus(200);
      expect(res.text).toMatchSnapshot();
    });
  });


  afterAll(async () => {
    await sequelize.drop({ force: true });
    await sequelize.close();
    server.close();
  });
});

