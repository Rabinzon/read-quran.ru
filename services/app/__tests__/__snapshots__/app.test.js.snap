// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app requests GET canonical surah page 1`] = `

<!DOCTYPE html>
<html lang="en">
  <head>
    <title>
      Священный Коран
    </title>
    <meta name="description"
          content="Сайт для удобного чтения Корана в хронологическом порядоке"
    >
    <meta name="keywords"
          content="Коран, читать коран, коран хронологический порядок"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="apple-touch-icon"
          sizes="180x180"
          href="/assets/favicon/apple-touch-icon.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="32x32"
          href="/assets/favicon/favicon-32x32.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="16x16"
          href="/assets/favicon/favicon-16x16.png?v=wAXXEkEEYG"
    >
    <link rel="manifest"
          href="/assets/favicon/site.webmanifest?v=wAXXEkEEYG"
    >
    <link rel="mask-icon"
          href="/assets/favicon/safari-pinned-tab.svg?v=wAXXEkEEYG"
          color="#5bbad5"
    >
    <link rel="shortcut icon"
          href="/assets/favicon/favicon.ico?v=wAXXEkEEYG"
    >
    <meta name="msapplication-TileColor"
          content="#da532c"
    >
    <meta name="msapplication-config"
          content="/assets/favicon/browserconfig.xml?v=wAXXEkEEYG"
    >
    <meta name="theme-color"
          content="#ffffff"
    >
    <script src="/assets/main.js">
    </script>
  </head>
  <body class="d-flex flex-column justify-content-between">
    <div class="nav-wrapper">
      <nav class="fixed-top navbar navbar-expand-lg navbar-light border-bottom">
        <a class="navbar-brand"
           href="/"
        >
          Священный Коран
        </a>
        <button class="navbar-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#navbarSupportedContent"
        >
          <span class="navbar-toggler-icon">
          </span>
        </button>
        <div class="collapse navbar-collapse"
             id="navbarSupportedContent"
        >
          <ul class="navbar-nav mr-auto">
          </ul>
          <form class="form-inline my-2 my-lg-0"
                url="/search"
                action="/search"
          >
            <input class="form-control mr-sm-2"
                   type="search"
                   name="q"
                   placeholder="Поиск"
                   aria-label="Поиск"
            >
            <button class="btn btn-outline-success my-2 my-sm-0"
                    type="submit"
            >
              Найти
            </button>
          </form>
        </div>
      </nav>
    </div>
    <main class="container">
      <div class="row justify-content-md-center">
        <div class="col-md-8 col-sm-12">
          <div class="my-3">
            <h1 class="mb-0"
                style="word-break: break-all;"
            >
              <small>
                96. «Сгусток»
              </small>
            </h1>
            <small>
              <i>
                хронолоческий порядок суры &nbsp;1
              </i>
            </small>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="col-md-8 col-sm-12 m-auto">
          <div class="ayat d-flex">
            <div class="d-block">
              <div class="badge badge-dark">
                96:1
              </div>
            </div>
            <div class="p-1"
                 style="font-size: 0.85rem;"
                 id="1"
            >
              (1). Читай! Во имя Господа твоего, который сотворил -
            </div>
          </div>
          <nav aria-label="...">
            <ul class="pagination pagination-sm d-flex justify-content-between mt-4">
              <li class="page-item disabled">
                <a class="page-link"
                   href="#"
                >
                  <span aria-hidden="true">
                    «
                  </span>
                  &nbsp; предыдущая
                </a>
              </li>
              <li class="page-item next">
                <a class="page-link"
                   href="/s/2"
                >
                  следующая &nbsp;
                  <span aria-hidden="true">
                    »
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <button class="btn btn-fixed js-to-top-btn d-none btn-dark btn-sm">
        наверх
      </button>
    </main>
    <footer class="footer p-3">
      <p class="text-right">
        <script type="text/javascript">
          (function (d, w, c) { (w[c] = w[c] || []).push(function() { try { w.yaCounter45770442 = new Ya.Metrika({ id:45770442, clickmap:true, trackLinks:true, accurateTrackBounce:true, webvisor:true, trackHash:true }); } catch(e) { } }); var n = d.getElementsByTagName("script")[0], s = d.createElement("script"), f = function () { n.parentNode.insertBefore(s, n); }; s.type = "text/javascript"; s.async = true; s.src = "https://mc.yandex.ru/metrika/watch.js"; if (w.opera == "[object Opera]") { d.addEventListener("DOMContentLoaded", f, false); } else { f(); } })(document, window, "yandex_metrika_callbacks");
        </script>
        <noscript>
          <div>
            <img src="https://mc.yandex.ru/watch/45770442"
                 style="position:absolute; left:-9999px;"
                 alt
            >
          </div>
        </noscript>
      </p>
      <div class="d-flex align-items-center justify-content-between">
        <div class="text-left">
          <span class="text-muted">
            <small>
              2017-2018 &nbsp;
            </small>
            <a href="/">
              <small>
                Коран в хронологическом порядке
              </small>
            </a>
          </span>
        </div>
        <a href="http://free-quran.ru/">
          <small>
            Бесплатный Коран
          </small>
        </a>
      </div>
    </footer>
  </body>
</html>

`;

exports[`app requests GET index page 1`] = `

<!DOCTYPE html>
<html lang="en">
  <head>
    <title>
      Священный Коран
    </title>
    <meta name="description"
          content="Сайт для удобного чтения Корана в хронологическом порядоке"
    >
    <meta name="keywords"
          content="Коран, читать коран, коран хронологический порядок"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="apple-touch-icon"
          sizes="180x180"
          href="/assets/favicon/apple-touch-icon.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="32x32"
          href="/assets/favicon/favicon-32x32.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="16x16"
          href="/assets/favicon/favicon-16x16.png?v=wAXXEkEEYG"
    >
    <link rel="manifest"
          href="/assets/favicon/site.webmanifest?v=wAXXEkEEYG"
    >
    <link rel="mask-icon"
          href="/assets/favicon/safari-pinned-tab.svg?v=wAXXEkEEYG"
          color="#5bbad5"
    >
    <link rel="shortcut icon"
          href="/assets/favicon/favicon.ico?v=wAXXEkEEYG"
    >
    <meta name="msapplication-TileColor"
          content="#da532c"
    >
    <meta name="msapplication-config"
          content="/assets/favicon/browserconfig.xml?v=wAXXEkEEYG"
    >
    <meta name="theme-color"
          content="#ffffff"
    >
    <script src="/assets/main.js">
    </script>
  </head>
  <body class="d-flex flex-column justify-content-between">
    <div class="nav-wrapper">
      <nav class="fixed-top navbar navbar-expand-lg navbar-light border-bottom">
        <a class="navbar-brand"
           href="/"
        >
          Священный Коран
        </a>
        <button class="navbar-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#navbarSupportedContent"
        >
          <span class="navbar-toggler-icon">
          </span>
        </button>
        <div class="collapse navbar-collapse"
             id="navbarSupportedContent"
        >
          <ul class="navbar-nav mr-auto">
          </ul>
          <form class="form-inline my-2 my-lg-0"
                url="/search"
                action="/search"
          >
            <input class="form-control mr-sm-2"
                   type="search"
                   name="q"
                   placeholder="Поиск"
                   aria-label="Поиск"
            >
            <button class="btn btn-outline-success my-2 my-sm-0"
                    type="submit"
            >
              Найти
            </button>
          </form>
        </div>
      </nav>
    </div>
    <main class="container">
      <div class="row justify-content-md-center">
        <div class="col-md-8 col-sm-12">
          <div class="text-center my-5">
            <h1>
              Священный Коран
            </h1>
            <p class="lead mb-0">
              в хронологическом порядке
            </p>
            <small>
              На данный момент доступен только перевод И. Ю. Крачковского.
            </small>
            <br>
            <a class="btn btn-link"
               href="/normal"
            >
              канонический порядок сур
            </a>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="col">
          <ol class="list">
            <li>
              <a class="p-1"
                 style="font-size: 0.85rem;"
                 href="/s/1"
              >
                Сгусток
              </a>
            </li>
          </ol>
        </div>
      </div>
      <button class="btn btn-fixed js-to-top-btn d-none btn-dark btn-sm">
        наверх
      </button>
    </main>
    <footer class="footer p-3">
      <p class="text-right">
        <script type="text/javascript">
          (function (d, w, c) { (w[c] = w[c] || []).push(function() { try { w.yaCounter45770442 = new Ya.Metrika({ id:45770442, clickmap:true, trackLinks:true, accurateTrackBounce:true, webvisor:true, trackHash:true }); } catch(e) { } }); var n = d.getElementsByTagName("script")[0], s = d.createElement("script"), f = function () { n.parentNode.insertBefore(s, n); }; s.type = "text/javascript"; s.async = true; s.src = "https://mc.yandex.ru/metrika/watch.js"; if (w.opera == "[object Opera]") { d.addEventListener("DOMContentLoaded", f, false); } else { f(); } })(document, window, "yandex_metrika_callbacks");
        </script>
        <noscript>
          <div>
            <img src="https://mc.yandex.ru/watch/45770442"
                 style="position:absolute; left:-9999px;"
                 alt
            >
          </div>
        </noscript>
      </p>
      <div class="d-flex align-items-center justify-content-between">
        <div class="text-left">
          <span class="text-muted">
            <small>
              2017-2018 &nbsp;
            </small>
            <a href="/">
              <small>
                Коран в хронологическом порядке
              </small>
            </a>
          </span>
        </div>
        <a href="http://free-quran.ru/">
          <small>
            Бесплатный Коран
          </small>
        </a>
      </div>
    </footer>
  </body>
</html>

`;

exports[`app requests GET search page 1`] = `

<!DOCTYPE html>
<html lang="en">
  <head>
    <title>
      Священный Коран
    </title>
    <meta name="description"
          content="Сайт для удобного чтения Корана в хронологическом порядоке"
    >
    <meta name="keywords"
          content="Коран, читать коран, коран хронологический порядок"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="apple-touch-icon"
          sizes="180x180"
          href="/assets/favicon/apple-touch-icon.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="32x32"
          href="/assets/favicon/favicon-32x32.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="16x16"
          href="/assets/favicon/favicon-16x16.png?v=wAXXEkEEYG"
    >
    <link rel="manifest"
          href="/assets/favicon/site.webmanifest?v=wAXXEkEEYG"
    >
    <link rel="mask-icon"
          href="/assets/favicon/safari-pinned-tab.svg?v=wAXXEkEEYG"
          color="#5bbad5"
    >
    <link rel="shortcut icon"
          href="/assets/favicon/favicon.ico?v=wAXXEkEEYG"
    >
    <meta name="msapplication-TileColor"
          content="#da532c"
    >
    <meta name="msapplication-config"
          content="/assets/favicon/browserconfig.xml?v=wAXXEkEEYG"
    >
    <meta name="theme-color"
          content="#ffffff"
    >
    <script src="/assets/main.js">
    </script>
  </head>
  <body class="d-flex flex-column justify-content-between">
    <div class="nav-wrapper">
      <nav class="fixed-top navbar navbar-expand-lg navbar-light border-bottom">
        <a class="navbar-brand"
           href="/"
        >
          Священный Коран
        </a>
        <button class="navbar-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#navbarSupportedContent"
        >
          <span class="navbar-toggler-icon">
          </span>
        </button>
        <div class="collapse navbar-collapse"
             id="navbarSupportedContent"
        >
          <ul class="navbar-nav mr-auto">
          </ul>
          <form class="form-inline my-2 my-lg-0"
                url="/search"
                action="/search"
          >
            <input class="form-control mr-sm-2"
                   value="читай"
                   type="search"
                   name="q"
                   placeholder="Поиск"
                   aria-label="Поиск"
            >
            <button class="btn btn-outline-success my-2 my-sm-0"
                    type="submit"
            >
              Найти
            </button>
          </form>
        </div>
      </nav>
    </div>
    <main class="container">
      <h5 class="mt-4">
        Количество найденных аятов по запросу&nbsp;
        <i>
          <strong>
            читай:&nbsp;
          </strong>
        </i>
        1
      </h5>
      <div class="mb-4">
        <small class="text-muted">
          Аяты отсортированы в хронологическом порядке.
        </small>
      </div>
      <div class="row">
        <div class="col">
          <ul class="pl-0 list">
            <small>
              <li class="my-1">
                <div class="badge badge-dark">
                  96:1
                </div>
                <a class="p-1 text-muted"
                   href="/s/1#1"
                >
                  (1). Читай! Во имя Господа твоего, который сотворил -
                </a>
              </li>
            </small>
          </ul>
        </div>
      </div>
      <button class="btn btn-fixed js-to-top-btn d-none btn-dark btn-sm">
        наверх
      </button>
    </main>
    <footer class="footer p-3">
      <p class="text-right">
        <script type="text/javascript">
          (function (d, w, c) { (w[c] = w[c] || []).push(function() { try { w.yaCounter45770442 = new Ya.Metrika({ id:45770442, clickmap:true, trackLinks:true, accurateTrackBounce:true, webvisor:true, trackHash:true }); } catch(e) { } }); var n = d.getElementsByTagName("script")[0], s = d.createElement("script"), f = function () { n.parentNode.insertBefore(s, n); }; s.type = "text/javascript"; s.async = true; s.src = "https://mc.yandex.ru/metrika/watch.js"; if (w.opera == "[object Opera]") { d.addEventListener("DOMContentLoaded", f, false); } else { f(); } })(document, window, "yandex_metrika_callbacks");
        </script>
        <noscript>
          <div>
            <img src="https://mc.yandex.ru/watch/45770442"
                 style="position:absolute; left:-9999px;"
                 alt
            >
          </div>
        </noscript>
      </p>
      <div class="d-flex align-items-center justify-content-between">
        <div class="text-left">
          <span class="text-muted">
            <small>
              2017-2018 &nbsp;
            </small>
            <a href="/">
              <small>
                Коран в хронологическом порядке
              </small>
            </a>
          </span>
        </div>
        <a href="http://free-quran.ru/">
          <small>
            Бесплатный Коран
          </small>
        </a>
      </div>
    </footer>
  </body>
</html>

`;

exports[`app requests GET surah page 1`] = `

<!DOCTYPE html>
<html lang="en">
  <head>
    <title>
      Священный Коран
    </title>
    <meta name="description"
          content="Сайт для удобного чтения Корана в хронологическом порядоке"
    >
    <meta name="keywords"
          content="Коран, читать коран, коран хронологический порядок"
    >
    <meta name="viewport"
          content="width=device-width, initial-scale=1"
    >
    <link rel="apple-touch-icon"
          sizes="180x180"
          href="/assets/favicon/apple-touch-icon.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="32x32"
          href="/assets/favicon/favicon-32x32.png?v=wAXXEkEEYG"
    >
    <link rel="icon"
          type="image/png"
          sizes="16x16"
          href="/assets/favicon/favicon-16x16.png?v=wAXXEkEEYG"
    >
    <link rel="manifest"
          href="/assets/favicon/site.webmanifest?v=wAXXEkEEYG"
    >
    <link rel="mask-icon"
          href="/assets/favicon/safari-pinned-tab.svg?v=wAXXEkEEYG"
          color="#5bbad5"
    >
    <link rel="shortcut icon"
          href="/assets/favicon/favicon.ico?v=wAXXEkEEYG"
    >
    <meta name="msapplication-TileColor"
          content="#da532c"
    >
    <meta name="msapplication-config"
          content="/assets/favicon/browserconfig.xml?v=wAXXEkEEYG"
    >
    <meta name="theme-color"
          content="#ffffff"
    >
    <script src="/assets/main.js">
    </script>
  </head>
  <body class="d-flex flex-column justify-content-between">
    <div class="nav-wrapper">
      <nav class="fixed-top navbar navbar-expand-lg navbar-light border-bottom">
        <a class="navbar-brand"
           href="/"
        >
          Священный Коран
        </a>
        <button class="navbar-toggler"
                type="button"
                data-toggle="collapse"
                data-target="#navbarSupportedContent"
        >
          <span class="navbar-toggler-icon">
          </span>
        </button>
        <div class="collapse navbar-collapse"
             id="navbarSupportedContent"
        >
          <ul class="navbar-nav mr-auto">
          </ul>
          <form class="form-inline my-2 my-lg-0"
                url="/search"
                action="/search"
          >
            <input class="form-control mr-sm-2"
                   type="search"
                   name="q"
                   placeholder="Поиск"
                   aria-label="Поиск"
            >
            <button class="btn btn-outline-success my-2 my-sm-0"
                    type="submit"
            >
              Найти
            </button>
          </form>
        </div>
      </nav>
    </div>
    <main class="container">
      <div class="row justify-content-md-center">
        <div class="col-md-8 col-sm-12">
          <div class="my-3">
            <h1 class="mb-0"
                style="word-break: break-all;"
            >
              <small>
                96. «Сгусток»
              </small>
            </h1>
            <small>
              <i>
                хронолоческий порядок суры &nbsp;1
              </i>
            </small>
          </div>
        </div>
      </div>
      <div class="row">
        <div class="col-md-8 col-sm-12 m-auto">
          <div class="ayat d-flex">
            <div class="d-block">
              <div class="badge badge-dark">
                96:1
              </div>
            </div>
            <div class="p-1"
                 style="font-size: 0.85rem;"
                 id="1"
            >
              (1). Читай! Во имя Господа твоего, который сотворил -
            </div>
          </div>
          <nav aria-label="...">
            <ul class="pagination pagination-sm d-flex justify-content-between mt-4">
              <li class="page-item disabled">
                <a class="page-link"
                   href="#"
                >
                  <span aria-hidden="true">
                    «
                  </span>
                  &nbsp; предыдущая
                </a>
              </li>
              <li class="page-item next">
                <a class="page-link"
                   href="/s/2"
                >
                  следующая &nbsp;
                  <span aria-hidden="true">
                    »
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </div>
      <button class="btn btn-fixed js-to-top-btn d-none btn-dark btn-sm">
        наверх
      </button>
    </main>
    <footer class="footer p-3">
      <p class="text-right">
        <script type="text/javascript">
          (function (d, w, c) { (w[c] = w[c] || []).push(function() { try { w.yaCounter45770442 = new Ya.Metrika({ id:45770442, clickmap:true, trackLinks:true, accurateTrackBounce:true, webvisor:true, trackHash:true }); } catch(e) { } }); var n = d.getElementsByTagName("script")[0], s = d.createElement("script"), f = function () { n.parentNode.insertBefore(s, n); }; s.type = "text/javascript"; s.async = true; s.src = "https://mc.yandex.ru/metrika/watch.js"; if (w.opera == "[object Opera]") { d.addEventListener("DOMContentLoaded", f, false); } else { f(); } })(document, window, "yandex_metrika_callbacks");
        </script>
        <noscript>
          <div>
            <img src="https://mc.yandex.ru/watch/45770442"
                 style="position:absolute; left:-9999px;"
                 alt
            >
          </div>
        </noscript>
      </p>
      <div class="d-flex align-items-center justify-content-between">
        <div class="text-left">
          <span class="text-muted">
            <small>
              2017-2018 &nbsp;
            </small>
            <a href="/">
              <small>
                Коран в хронологическом порядке
              </small>
            </a>
          </span>
        </div>
        <a href="http://free-quran.ru/">
          <small>
            Бесплатный Коран
          </small>
        </a>
      </div>
    </footer>
  </body>
</html>

`;
